schema/impl.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/GoogleCloudPlatform/heapster/sinks/cache"
+)
+
+var lock sync.RWMutex
```

Comment:
@vishh @rjnagal The lock was moved outside the internal cluster object in commit https://github.com/afein/heapster/commit/32b54bd90a9f4a731ed5f449f9ca617e098a70eb , after  comment https://github.com/GoogleCloudPlatform/heapster/pull/358#discussion_r32685465 and the followup offline discussion.

Due to a dependency with the upcoming [Schema 3] PR, the lock will be re-inserted into the realCluster as part of [Schema 3].